controller/deployments.go

```diff
@@ -381,6 +384,67 @@ func (c *DeploymentsController) ShowSpaceEnvironments(ctx *app.ShowSpaceEnvironm
 	return ctx.OK(res)
 }
 
+// WatchEnvironmentEvents runs the watchEnvironmentEvents action.
+func (c *DeploymentsController) WatchEnvironmentEvents(ctx *app.WatchEnvironmentEventsDeploymentsContext) error {
+	c.connectEventsHandler(ctx).ServeHTTP(ctx.ResponseWriter, ctx.Request)
+	return nil
+}
+
+func (c *DeploymentsController) connectEventsHandler(ctx *app.WatchEnvironmentEventsDeploymentsContext) websocket.Handler {
+	return func(ws *websocket.Conn) {
+		defer ws.Close()
+
+		kc, err := c.GetKubeClient(ctx)
+		defer cleanup(kc)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"err": err,
+			}, "error accessing Auth server")
+			return
+		}
+
+		store, stopWs := kc.WatchEventsInNamespace(ctx.EnvName)
+		defer close(stopWs)
+
+		go func() {
+			for {
+				var m string
+				err := websocket.Message.Receive(ws, &m)
+				if err != nil {
+					if err != io.EOF {
+						log.Error(ctx, map[string]interface{}{
+							"err": err,
+						}, "error reading from websocket")
+					}
+					store.Close()
+					return
+				}
+			}
+		}()
+
+		for {
+			item, err := store.Pop(cache.PopProcessFunc(func(item interface{}) error {
+				return nil
+			}))
+			if err != nil {
+				if err != cache.FIFOClosedError {
+					log.Error(ctx, map[string]interface{}{
+						"err": err,
+					}, "error receiving events: "+time.Now().String())
```

Comment:
I think the timestamp here is redundant, the WIT logger adds a timestamp to all messages.